extensions/modules/mail/src/main/java/org/exist/xquery/modules/mail/MessageListFunctions.java

```diff
@@ -557,227 +544,166 @@ private SearchTerm[] parseChildSearchTerms( Node terms ) throws XPathException
-			throw( new XPathException(this, "value not specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
+		if ( value.isEmpty() ) {
+			throw new XPathException(this, "value not specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) ;
 		}
-		
-		if( StringUtils.isEmpty(format) ) {
-			throw( new XPathException(this, "format not specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
+
+		String format = ((Element)terms).getAttribute( "format" );
+		if ( format.isEmpty() ) {
+			throw new XPathException(this, "format not specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) ;
 		}
 		
 		int  cp = parseComparisonAttribute( terms );
-		
 		try {
 			SimpleDateFormat sdf = new SimpleDateFormat( format );
-			
 			Date date = sdf.parse( value );
-			
-			st = new ReceivedDateTerm( cp, date );
+			return new ReceivedDateTerm( cp, date );
 		}
 		catch( ParseException pe ) {
 			throw( new XPathException(this, "Cannot parse date value: " + value + ", using format: " + format + ", for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
 		}
-		
-		return( st );
 	}
 
 	private int parseComparisonAttribute( Node terms ) throws XPathException
 	{
-		int  cp = ComparisonTerm.EQ;
-		
 		String comp  = ((Element)terms).getAttribute( "comparison" );
-		
-		if( comp != null && !comp.isEmpty()) {
-			if( "eq".equalsIgnoreCase(comp) ) {
-				cp = ComparisonTerm.EQ;
-			} else if( "ge".equalsIgnoreCase(comp) ) {
-				cp = ComparisonTerm.GE;
-			} else if( "gt".equalsIgnoreCase(comp) ) {
-				cp = ComparisonTerm.GT;
-			} else if( "le".equalsIgnoreCase(comp) ) {
-				cp = ComparisonTerm.LE;
-			} else if( "lt".equalsIgnoreCase(comp) ) {
-				cp = ComparisonTerm.LT;
-			} else if( "ne".equalsIgnoreCase(comp) ) {
-				cp = ComparisonTerm.NE;
-			} else {
-				throw( new XPathException(this, "Invalid comparison: " + comp + ", for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
-			}
+		if (comp.isEmpty()) {
+			throw new XPathException(this, "comparison attribute must be specified for term with type: " + ((Element) terms).getAttribute("type"));
+		}
+		int  cp = ComparisonTerm.EQ;
+		if ("eq".equalsIgnoreCase(comp) ) {
+			cp = ComparisonTerm.EQ;
+		} else if( "ge".equalsIgnoreCase(comp) ) {
+			cp = ComparisonTerm.GE;
+		} else if( "gt".equalsIgnoreCase(comp) ) {
+			cp = ComparisonTerm.GT;
+		} else if( "le".equalsIgnoreCase(comp) ) {
+			cp = ComparisonTerm.LE;
+		} else if( "lt".equalsIgnoreCase(comp) ) {
+			cp = ComparisonTerm.LT;
+		} else if( "ne".equalsIgnoreCase(comp) ) {
+			cp = ComparisonTerm.NE;
 		} else {
-			throw( new XPathException(this, "comparison attribute must be specified for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
+			throw( new XPathException(this, "Invalid comparison: " + comp + ", for term with type: " + ((Element)terms).getAttribute( "type" ) ) );
 		}
-		
-		return( cp );
+		return cp;
```

Comment:
.. switch as function

Reply:
I propose to change this in a followup PR